app/controllers/application_controller.rb

```diff
@@ -13,4 +18,23 @@ def jsonapi_render(collection, filters)
       end
     end
   end
+
+  def set_user
+    @current_user = User.from_jwt Authenticator.from_token auth_token
+  end
+
+  def set_roles
+    return unless current_user
+    current_user.roles = panoptes_client.roles current_user.id
+  end
+
+  def panoptes_client
+    @panoptes_client ||= PanoptesClient.new auth_token
+  end
+
+  def auth_token
+    return @auth_token if @auth_token
+    authorization = request.headers['Authorization']
```

Comment:
super minor feel free to dismiss, you can use the assignment expression to capture the clause on `auth.present?` and save an operation / branch test.
```
if (authorization = request.headers['Authorization'])
  @auth_token = authorization.sub(/^Bearer /, '')
end
```

I would probably just collapse them all to a regex with a capture group
`@auth_token = authorization.match(/\ABearer\s(.+)\z/)[1] # 2nd index is the match capture group`